Source/Dafny/Verifier/Translator.cs

```diff
@@ -3553,10 +3552,15 @@ class Specialization
       var synonyms = Bpl.Expr.Eq(
         funcAppl,
         ModeledAsBoxType(f.ResultType) ? BoxIfUnboxed(overridingFuncAppl, overridingFunction.ResultType) : overridingFuncAppl);
-
+      // add overridingFunction#canCall ==> f#canCall to the axiom
+      Bpl.IdentifierExpr callName = new Bpl.IdentifierExpr(f.tok, f.FullSanitizedName + "#canCall", Bpl.Type.Bool);
+      Bpl.Expr canCallFunc = new Bpl.NAryExpr(f.tok, new Bpl.FunctionCall(callName), argsJF);
+      callName = new Bpl.IdentifierExpr(overridingFunction.tok, overridingFunction.FullSanitizedName + "#canCall", Bpl.Type.Bool);
+      Bpl.Expr canCallOverridingFunc = new Bpl.NAryExpr(f.tok, new Bpl.FunctionCall(callName), argsCF);
```

Comment:
You can leave off the types of the declared variables, which (I find) makes the code easier to read.
```suggestion
      var callName = new Bpl.IdentifierExpr(f.tok, f.FullSanitizedName + "#canCall", Bpl.Type.Bool);
      var canCallFunc = new Bpl.NAryExpr(f.tok, new Bpl.FunctionCall(callName), argsJF);
      callName = new Bpl.IdentifierExpr(overridingFunction.tok, overridingFunction.FullSanitizedName + "#canCall", Bpl.Type.Bool);
      var canCallOverridingFunc = new Bpl.NAryExpr(f.tok, new Bpl.FunctionCall(callName), argsCF);
```